src/ui/public/modals/confirm_modal.js

```diff
@@ -72,22 +68,33 @@ module.factory('confirmModal', function ($rootScope, $compile) {
       options.onClose();
     };
 
-    const modalInstance = $compile(template)(confirmScope);
-    modalPopover = new ModalOverlay(modalInstance);
+    function showModal(confirmScope) {
+      const modalInstance = $compile(template)(confirmScope);
+      modalPopover = new ModalOverlay(modalInstance);
+      angular.element(document.body).on('keydown', (event) => {
+        if (event.keyCode === 27) {
+          confirmScope.onCancel();
+        }
+      });
 
-    angular.element(document.body).on('keydown', (event) => {
-      if(event.keyCode === 27) {
-        confirmScope.onCancel();
-      }
-    });
+      modalInstance.find('[data-test-subj=confirmModalConfirmButton]').focus();
+    }
 
-    modalInstance.find('[data-test-subj=confirmModalConfirmButton]').focus();
+    if (modalPopover) {
+      confirmQueue.unshift(confirmScope);
```

Comment:
Pop takes off the last element in the array, and push pushes it on the end, so push/pop is LIFO.  unshift/pop is FIFO which is what I was aiming for, to preserve the order of the calls.